packages/lisk-p2p/src/p2p_types.ts

```diff
@@ -21,6 +22,46 @@ export interface IP2PResponsePacket {}
 
 export interface IP2PNodeStatus {}
 
+export interface Logger {
+	readonly error: (message: string) => void;
+	readonly info: (message: string) => void;
+	readonly log: (message: string) => void;
+	readonly trace: (message: string) => void;
+	readonly warn: (message: string) => void;
+}
+
+// ---- Start IP2PConfig interfaces ----
+
+export interface P2PConfigPeersAccess {
+	readonly blacklist: ReadonlyArray<string>;
+}
+
+export interface P2PConfigPeersOptions {
+	readonly broadhashConsensusCalculationInterval: number;
+	readonly timeout: number;
+	readonly wsEngine: string;
+}
+
+export interface P2PConfigPeers {
+	readonly access: P2PConfigPeersAccess;
+	readonly enabled: boolean;
+	readonly list: ReadonlyArray<string>;
+	readonly options: P2PConfigPeersOptions;
+}
+
+export interface P2PConfig {
+	readonly logger?: Logger;
```

Comment:
do we need to pass logger?
I'm not sure if library should support expecting logger, socket-cluster have possibility to show logs but not inserting logger for example
we could use https://www.npmjs.com/package/debug?

Reply:
We need to account for multiple processes. This means that we need some way to aggregate the logs from different libraries and modules. I don't know if `debug` support this. It depends on the controller. Also there is https://github.com/trentm/node-bunyan and https://github.com/winstonjs/winston to consider.

Reply:
We have decided to use even emitting for the non-functional actions, and if needed, we will use `debug` internally, but not exposing the logs to the users